File: product-services/src/main/java/com/microservices/productservices/service/CategoryService.java
package com.microservices.productservices.service;

import com.microservices.productservices.payload.request.CategoryRequest;
import com.microservices.productservices.payload.response.CategoryResponse;

import java.util.List;
import java.util.UUID;

public interface CategoryService {

    CategoryResponse create(CategoryRequest categoryRequest);

    void setImage(UUID id, String image);

    void switchStatus(UUID id);

    void trash(UUID id);

    void isDisplay(UUID id);

    CategoryResponse getById(UUID id);

    List<CategoryResponse> getAll();

    CategoryResponse update(UUID id, CategoryRequest categoryRequest);

    CategoryResponse delete(UUID id);

    
}

File: reactjs-app/src/services/CategoryService.js
import httpAxios from '../httpAxios';

const CategoryService = {
    create: (data) => {
        return httpAxios.post(`product-services/api/categories/create`, data);
    },
    setImage: (data) => {
        return httpAxios.put(`product-services/api/categories/set-image`, data);
    },
    getById: (id) => {
        return httpAxios.get(`product-services/api/categories/get-by-id/${id}`);
    },
    getByUser: (userId) => {
        return httpAxios.get(`product-services/api/categories/get-by-user/${userId}`);
    },
    getAll: () => {
        return httpAxios.get(`product-services/api/categories/get-all`);
    },
    update: (id, data) => { 
        return httpAxios.put(`product-services/api/categories/update/${id}`, data);
    },
    sitchStatus: (id) => {
        return httpAxios.put(`product-services/api/categories/switch-status/${id}`);
    },
    trash: (id) => {
        return httpAxios.put(`product-services/api/categories/trash/${id}`);
    },
    display: (id) => {
        return httpAxios.put(`product-services/api/categories/display/${id}`);
    },
    delete: (id) => {
        return httpAxios.delete(`product-services/api/categories/delete/${id}`);
    },
}
export default CategoryService;
